doc/design/if_else_op.md

```diff
@@ -0,0 +1,55 @@
+In an if_op, only inputs with condition satisfied will be run. The op could have multiple inputs and multiple outputs.
+We should have the following design:
+
+```python
```

Comment:
Basing on https://github.com/PaddlePaddle/Paddle/pull/3827, Python bindings of operators and layers should return Vars. So I think that this usage could be

```python
x1 = Var()
x2 = Var()
y1 = Var()
y2 = Var()
cond = Var()

with paddle.block() as left:
  x1 = if_input(x1)
  x2 = if_input(x2)
  y1 = mul(x1, x2)
  y2 = add(x1, x2)

with paddle.block() as right:
  x1 = if_input(x1)
  x2 = if_input(x2)
  y1 = add(x1, x2)
  y2 = mul(x1, x2)

y1, y2 = layer.if(cond, left, right) # This line is equal to layer.if(cond, left, right)

Reply:
```python
import paddle as pd

x1 = Var()
x2 = Var()
y1 = Var()
y2 = Var()
cond = Var()

c1 = pd.IfElseOp(cond, inputs=[x1, x2], outputs=[y1, y2])
with c1.true_block():
    x1, x2 = c1.inputs
    y1, y2 = c1.outputs

    y1 = mul(x1, x2)
    y2 = add(x1, x2)

with c1.false_block():
    x1, x2 = c1.inputs
    y1, y2 = c1.outputs

    y1 = add(x1, x2)
    y2 = mul(x1, x2)
```

@wangkuiyi

Reply:
I believe @Superjom 's proposal is better than mine!

Reply:
[Above example](https://github.com/PaddlePaddle/Paddle/pull/3828#discussion_r136707032) breaks the convention that Python operator bindings must return Vars.  

However, [mine](https://github.com/PaddlePaddle/Paddle/pull/3828#discussion_r136706744) has a severe problem that the two branches would overwrite y1 and y2. A right solution should keep two copies of y1 and y2 for the left and the right branches respectively and then merge these two copies into the global variables y1 and y2.

A correct solution can be derived easily from @Superjom 's solution -- to rename `paddle.IfElseOp` into `paddle.CreateIfElseOpBuilder`.

Reply:
```python
import paddle as pd

x1 = Var()
x2 = Var()
cond = Var()
c = pd.create_ifelseop_builder(inputs=[x1, x2], output_num=2)
with c.true_block() as b:
    x1, x2 = b.inputs()
    b.set_output(0, mul(x1, x2))
    b.set_output(1, add(x1, x2))

with c.false_block() as b:
    x1, x2 = c1.inputs()
    b.set_output(0, add(x1, x2))
    b.set_output(1, mul(x1, x2))

out1, out2 = c(cond)
```